ops/terraform/modules/resources/s3_pii/templates/bucket-policy.json

```diff
@@ -1,62 +1,48 @@
+${jsonencode(
 {
   "Version": "2012-10-17",
   "Id": "bfd-${env}-${bucket_name}-s3-policy",
   "Statement": [
-    {
-      "Sid": "BFDPIIS3ListBucket",
-      "Effect": "Allow",
-      "Principal": {
-        "AWS": [
-          %{ for reader in readers ~}"${reader}",%{ endfor ~}
-          %{ for writer in writers ~}"${writer}",%{ endfor ~}
-          %{ for admin in admins ~}"${admin}",%{ endfor ~}
-          "${root}"
-        ]
-      },
-      "Action": [
-        "s3:ListBucket"
-      ],
-      "Resource": "arn:aws:s3:::${bucket_id}"
-    },
     {
       "Sid": "BFDPIIS3ReadBucket",
       "Effect": "Allow",
       "Principal": {
         "AWS": [
-          %{ for reader in readers ~}"${reader}",%{ endfor ~}
-          %{ for writer in writers ~}"${writer}",%{ endfor ~}
-          %{ for admin in admins ~}"${admin}",%{ endfor ~}
-          "${root}"
+          for arn in concat(readers, writers) : "${arn}"
         ]
       },
       "Action": [
-        "s3:GetObject"
+        "s3:ListBucket*",
```

Comment:
Suggest that you could add `s3:HeadBucket`, and `s3: GetBucketLocation` to the actions list.

Reply:
It appears `s3:HeadBucket` is not a valid operation for a specific S3 bucket, only for `*` so not valid within the context of the S3 policy.  The IAM writer policy already has `s3:HeadBucket`.  I have added `s3:GetBucket*` to the S3 policy to be more inclusive of read operations.